sdk/src/main/java/io/opentelemetry/sdk/metrics/MetricData.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import com.google.auto.value.AutoValue;
+import io.opentelemetry.trace.Timestamp;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+@AutoValue
+public abstract class MetricData {
+  MetricData() {}
+
+  /**
+   * Returns the {@link MetricDescriptor} of this metric.
+   *
+   * @return the {@code MetricDescriptor} of this metric.
+   * @since 0.1.0
+   */
+  public abstract MetricDescriptor getMetricDescriptor();
+
+  /**
+   * Returns the type of this metric.
+   *
+   * @return the type of this metric.
+   * @since 0.1.0
+   */
+  public abstract Type getType();
```

Comment:
Why type is not part of descriptor?

Reply:
Thought more, makes sense to have it in the descriptor.